src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -422,6 +397,23 @@ static VsTextSpan GetVsTextSpan(SourceText text, int position, int virtualSpace)
         private static int GetPositionWithinDocumentBounds(int position, int documentLength)
             => Math.Min(documentLength, Math.Max(position, 0));
 
+        private static VsTextSpan GetVsTextSpan(SourceText text, TextSpan textSpan, bool allowInvalidSpan)
```

Comment:
Not your doing, but oh goodness is a parameter called "allowInvalidSpan" terrifying.

Reply:
yeah.  for context.  this allows us to use 'cached navto results from previous vs session' without having to do the costly work to ensure they're valid.  IN a sense the feature *is* allowing navigation to invalid locations, albeit, with a message to the user that this may be happening.  it enables huge speedups, but it means we need to be resilient to old data being stale and innacurate.